### PR TITLE
DoubleTap Support

### PR DESCRIPTION
**Pull request by rafael-chavez**
## Change list

DoubleTap support to TouchAction
## Types of changes

What types of changes are you proposing/introducing to Java client?
_Put an `x` in the boxes that apply_
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by rafael-chavez** [comment]
@TikhomirovSergey @SrinivasanTarget

**Comment by SrinivasanTarget** [comment]
Awesome. Can you add some tests please? Will review soon.

**Comment by TikhomirovSergey** [comment]
tests
+1

File: src/test/java/io/appium/java_client/ios/IOSGesturesTest.java [hunk 1]
```diff
@@ -33,6 +34,14 @@
         driver.tap(2, e, 2000);
         assertEquals(driver.findElementByXPath("//*[@name = \"Answer\"]").getText(), "6");
     }
+    
+    @Test public void doubleTapTest() {
+        IOSElement firstField = (IOSElement) driver.findElementById("IntegerA");
+        firstField.sendKeys("2");
+        driver.doubleTap(firstField);
+        IOSElement editingMenu = (IOSElement) driver.findElementByClassName("UIAEditingMenu");
+        assertNotNull(editingMenu);
```
Comment (rafael-chavez) [inline src/test/java/io/appium/java_client/ios/IOSGesturesTest.java]: there was little to double tap on in ios TestApp. But found the editing menu shows up when you double tap on one of the fields.

**Comment by rafael-chavez** [comment]
added android test as well!

File: src/test/java/io/appium/java_client/android/AndroidGestureTest.java [hunk 2]
```diff
@@ -40,6 +42,15 @@
             .findElementById("io.appium.android.apis:id/button_toggle").getText());
     }
 
+    @Test public void doubleTapTest() throws Exception {
+        driver.startActivity("io.appium.android.apis", ".view.TextSwitcher1");
+        String textBeforeDoubleTap = (driver.findElementByClassName("android.widget.TextView")).getText();
+        AndroidElement next = driver.findElementById("io.appium.android.apis:id/next");
```
Comment (rafael-chavez) [inline src/test/java/io/appium/java_client/android/AndroidGestureTest.java]: next is used lol

**Comment by SrinivasanTarget** [comment]
@rafael-chavez Looks good but some remarks,

1)Correct java doc here: https://github.com/appium/java-client/pull/444/files#diff-970f723bf2aa8111d5817a3de004cc13R208 & https://github.com/appium/java-client/pull/444/files#diff-6f4ca47e6c9b3f6056a3d5f88a6d7821R65
2)please change the java doc here: https://github.com/appium/java-client/pull/444/files#diff-c60f84c0cc90f37a9a4513eeebd4896dR400 like this ```@see TouchShortcuts#doubleTap(WebElement, int, int)```. Here too https://github.com/appium/java-client/pull/444/files#diff-c60f84c0cc90f37a9a4513eeebd4896dR408.
3)Few Checkstyle errors: https://gist.github.com/SrinivasanTarget/d44880040b29b67ff8861fca1409be74

**Comment by rafael-chavez** [comment]
i will update with the javadoc and style fixes.

**Comment by rafael-chavez** [comment]
Fixed java doc and codestyle @SrinivasanTarget !

**Comment by SrinivasanTarget** [comment]
@rafael-chavez Does the tests passed? I see server errors: https://gist.github.com/SrinivasanTarget/1a5683fc5523bd50b74a98884cf484b3

**Comment by rafael-chavez** [comment]
that error makes sense @SrinivasanTarget as i only added support for iOS driver.

**Comment by SrinivasanTarget** [comment]
@rafael-chavez Yes i see that there is no support for ```doubleTap``` in ```appium-android-driver```. I would suggest to move this to ```IOSDriver.java``` instead of  ```AppiumDriver.java``` as there is no support. What do you say @TikhomirovSergey ?

**Comment by rafael-chavez** [comment]
Im ok with either of these two suggestions @SrinivasanTarget @TikhomirovSergey

**Comment by rafael-chavez** [comment]
Im moving this to IOSDriver if thats ok with you @SrinivasanTarget

**Comment by rafael-chavez** [comment]
@SrinivasanTarget @TikhomirovSergey moved to IOSDriver and removed Android test added.

**Comment by rafael-chavez** [comment]
ping @SrinivasanTarget @TikhomirovSergey

**Comment by SrinivasanTarget** [comment]
@rafael-chavez Apologies for the delay. I'm on vacation till 12th Sep. We were working on splitting TouchActions as you can see here, #458. We will get this in soon. @TikhomirovSergey  FYI

**Pull request closed without merging**